ops/model.py

```diff
@@ -552,24 +650,29 @@ class Binding:
         name: The name of the endpoint this binding represents (eg, 'db')
     """
 
-    def __init__(self, name, relation_id, backend):
+    def __init__(self, name: str, relation_id: Optional[int], backend: '_ModelBackend'):
         self.name = name
         self._relation_id = relation_id
         self._backend = backend
         self._network = None
 
+    def _network_get(self, name: str, relation_id: Optional[int] = None) -> 'Network':
+        raw_data = self._backend.network_get(name, relation_id)
+        network_dict = typing.cast(_NetworkDict, raw_data)
+        return Network(network_dict)
```

Comment:
why doesn't `_ModelBackend.network_get` just have a typed return value so you don't need the cast here?

Reply:
done! good catch. No idea why pyright didn't whine about network_get not having a return type